packages/tooltip/src/useTooltip.ts

```diff
@@ -5,16 +5,34 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useState, useEffect, useRef } from 'react';
+import { useState, useEffect, useRef, ButtonHTMLAttributes } from 'react';
 import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
 
-export function useTooltip({ delayMilliseconds = 500, id, isVisible } = {}) {
+export interface IUseTooltipProps {
+  delayMilliseconds?: number;
+  id?: string;
+  isVisible?: boolean;
+}
+
+export interface IUseTooltipReturnValue {
+  isVisible?: boolean;
+  getTooltipProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getTriggerProps: <T>(options?: T) => T & React.ButtonHTMLAttributes<any>;
+  openTooltip: (delayMs?: number) => void;
+  closeTooltip: (delayMs?: number) => void;
+}
+
+export const useTooltip = ({
+  delayMilliseconds = 500,
+  id,
+  isVisible
+}: IUseTooltipProps = {}): IUseTooltipReturnValue => {
   const [visibility, setVisibility] = useState(isVisible);
   const [_id] = useState(id || generateId('garden-tooltip-container'));
   const isMounted = useRef(false);
 
-  let openTooltipTimeout;
-  let closeTooltipTimeout;
+  let openTooltipTimeout: NodeJS.Timeout = setTimeout(() => {}, 0);
```

Comment:
Let's not use `NodeJS` type definitions in front-end code. Why do these need a default value?

Reply:
> Let's not use NodeJS type definitions in front-end code. 

Oh yes, this is a good point! I thought this code might have been in a testing environment running in NodeJS because VSCode inferred the type as `NodeJS.Timeout`. This is my bad!

![Screen Shot 2019-11-01 at 10 30 46 AM](https://user-images.githubusercontent.com/1811365/68043986-cc549c80-fc93-11e9-9302-7b285652d1a8.png)

> Why do these need a default value?

We need default values because TypeScript warns that the variables are used before being assigned:

![Screen Shot 2019-11-01 at 10 31 26 AM](https://user-images.githubusercontent.com/1811365/68044135-f60dc380-fc93-11e9-9365-33ec754cd2fe.png)

**Given the feedback, I decided to remove `NodeJS.Timeout` and the default value of `setTimeout(() => {}, 0)` in favor having the initial values be `undefined`. That change can be seen in [this commit](https://github.com/zendeskgarden/react-containers/pull/124/commits/0436aea8cf3ad50f088f5af2d2680d5aaca24414). Let me know what you think 👍.**